docs/Windows.md

```diff
@@ -1,43 +1,172 @@
+
+Those installations to not add any values to the PATH environment variable. The PATH environment variable has to adjusted separatly (see below).
+
+The directories created by those installations are as follows:
+
+- windows-toolchain-amd64.msi: `C:\Library\Developer\Toolchains\unknown-Asserts-development.xctoolchain`
+- windows-sdk.msi: `C:\Library\Developer\Platforms\Windows.platform`
+- windows-runtime-amd64.msi: `C:\Library\Swift\Current\bin`
+
+#### Further requirements: CMake
+
+On Windows, the most convenient setup for building Swift projects currently involves the use of CMake.  This requires CMake 3.15+ for Swift support.  You can download CMake from <https://cmake.org/>.
+
+In the following, we assume you added CMake to the C:\Libary folder, so that the CMake executable is in `C:\Library\cmake\bin`.
+
+#### Further requirements: ICU
+
+You will need the ICU libraries from <http://site.icu-project.org/>. The nightlies are built against ICU 64.2 from the ICU project. You can download the binaries for that from http://download./files/icu4c/64.2/icu4c-64_2-Win64-MSVC2017.zip.
 In these instructions we assume you rename the extracted icu folder `icu4c-64_2-Win64-MSVC2017` to `icu-64.2` and move it to `C:\Library` and that you rename `bin64` to `bin`.
 
-#### Building and running swift code
+_Note:_ Currently you also need `icu-63.1`.
+
+#### Building Swift code
+
+You should use a CMake project to build a Swift program. An example CMake project with support for mixed-language support is available at <https://github.com/compnerd/swift-cmake-demo>.
+
+For building such a CMake project the following paths have to be added to the PATH environment variable (adjust those paths to your actual installation folders):
+
+- `C:\Library\Developer\Toolchains\unknown-Asserts-development.xctoolchain\usr\bin`
+- `C:\Library\cmake\bin`
+- `C:\Library\icu-64.2\bin`
+
+_Note:_ Currently, `C:\Library\icu-63.1\bin` has to be added, too.
+
+You can choose to add those paths permanently to your PATH environment variable, or you formulate a Windows batch script that first adds those paths to the PATH environment variable and then calls the build commands which are listed below. Be sure to add those paths before other paths, so e.g. use
 
-1. To have the required Visual Studio toolchain available ensure you use the `x64 Native Tools Command Prompt for VS 2017` that came with Visual Studio. Search inside the Visual Studio application folder to find it.
-2. You need to add the folder containing `swiftc.exe` to the `PATH` environment variable to use the toolchain.
-3. You also need to add the folder containing the icu dlls to the `PATH` environment variable. 
+SET PATH=%SCRIPT_DIR%\cmake\bin;%PATH%
```

Comment:
Please wrap that in a verbatim command block.  See the rest of the document for the use of the triple backticks.